### PR TITLE
IA-1812: Org unit form multiple states

### PR DESCRIPTION
**Pull request by codeKameleon**
Original issue: adding an alias cleared the org unit tree view input 
Finally this part of the code as to be totally refactored as the org unit form was using one for state and orgUniInfo too

We passed everything to Typescript in order to help with this mix-up of state (that was creating the original issue) 

Related JIRA tickets : [IA-1812](https://bluesquare.atlassian.net/browse/IA-1812)

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests
## Changes

- migrate `OrgUnitInfos` to typescript
- migrate `OrgUnitIForm` to typescript
- remove unnecessary form state in `OrgUnitInfos`
- split `OrgUnitAction` into separate file
- Wrote cypress test to test org unit creation

## How to test

Go to Orgunits > Create 
Then  select a parent orgunit and try adding an alias
Adding an alias should not empty the parent orgunit treeview

You should also try to edit and save an existing org unit

## Print screen / video

[recording-2023-01-13-16-07-51.webm](https://user-images.githubusercontent.com/25134301/212352559-52635d01-4ab8-4e75-84ec-dfbadd0e9519.webm)

[IA-1812]: https://bluesquare.atlassian.net/browse/IA-1812?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitForm.js [hunk 1]
```diff
@@ -82,8 +82,6 @@ export const OrgUnitForm = ({
             const actualAliases = value.filter(alias => alias !== '');
             if (newAlias !== '' && !isEqual(actualAliases, orgUnitAliases)) {
                 setOrgUnitModified(true);
-            } else {
-                setOrgUnitModified(false);
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitForm.js:86]: This is not fixing the issue in fact, this is avoiding it. Can you have a look into `OrgUnitInfosComponent` please. The `reformatOrgUnit` method looks weird. It's using `orgunit.parent` but we only have a `orgunit.parent_id`

**Review by beygorghor — CHANGES_REQUESTED** [review]

**Review by beygorghor — APPROVED** [review]

**Comment by beygorghor** [comment]
Added cypress test on creation.
@cypress

File: hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitInfos.tsx [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+}));
+
+type ParentOrgUnit = {
+    id: number;
+    name: string;
+    source: string;
+    source_id: number;
+    parent: ParentOrgUnit;
+    parent_name: string;
+};
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitInfos.tsx]: Is it not a sub type of org unit?

File: hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitForm.tsx [hunk 3]
```diff
@@ -1,58 +1,80 @@
...
-        groups: orgUnit.groups?.map(g => g.id) ?? [],
-        sub_source: orgUnit.sub_source,
-        validation_status: orgUnit.validation_status,
-        aliases: orgUnit.aliases,
-        parent_id: orgUnit.parent_id,
-        source_ref: orgUnit.source_ref,
-        creator: orgUnit.creator,
-    };
-};
+const initialFormState = orgUnit => ({
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitForm.tsx]: no typing?

**Review by quang-le — APPROVED** [review]
much cleaner that way. works on my machine

**Pull request merged**